iphone/cli/commands/_buildModule.js

```diff
@@ -1078,17 +1081,25 @@ iOSModuleBuilder.prototype.runModule = function runModule(cli, next) {
 		function (cb) {
 			// 6. run the app
 			this.logger.debug(__('Running example project...', tmpDir.cyan));
-			runTiCommand(
-				[
-					'build',
-					'-p', 'ios',
-					'-d', tmpProjectDir,
-					'--no-prompt',
-					'--no-colors',
-					'--no-progress-bars'
-				],
-				cb
-			);
+			const buildArgs = [
+				'build',
+				'-p', 'ios',
+				'-d', tmpProjectDir,
+				'--no-prompt',
+				'--no-colors',
+				'--no-progress-bars'
+			];
+
+			if (this.target) {
+				buildArgs.push('-T');
+				buildArgs.push(this.target);
+			}
+			if (this.deviceId) {
+				buildArgs.push('-C');
+				buildArgs.push(this.deviceId);
```

Comment:
non-blocking nitpick:

```suggestion
				buildArgs.push('-C', this.deviceId);
```